### PR TITLE
drivers: add driver for InvenSense ITG-320X 3-axis gyroscope

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR adds a driver for the InvenSense ITG320X 3-axis digital gyroscope. The driver can be used with InvenSense ITG3200, ITG3205, and MPU3050. The latter one can be used only with reduced feature set.

ITG320X can be found for example on the widespreaded GY-85 9-DOF (Accelerometer ADXL345, Gyroscope ITG3200, Magnetometer HMC5883) module. Together with the existing ```adxl345``` driver and the ```hmc5883l``` driver contributed in PR #10083 it becomes possible to use the whole 9-DOF module

The driver implements

- basic polling mechanism
- interrupt handling
- power-down/up functions
- SAUL interface

### Testing procedure

This PR includes ```tests/driver_itg320x``` which can be used for testing. This test application demonstrates the usage of the driver and shows the different approaches to fetch data, by using either

- periodically fetching the data, or
- fetching the data when the data-ready interrupt is triggered.

To use the latter approach, the according GPIO to which the sensor's **INT** output pin is connected has to be configured and initialized by the application. In this test application, this is done by defining ```USE_ITG320X_DRDY``` and overrding the default configuration parameter ```ITG320X_PARAM_INT_PIN``` if necessary, for example:

```
CFLAGS="-DUSE_ITG320X_DRDY -DITG320X_PARAM_INT_PIN=\(GPIO_PIN\(0,3\)\)" \
make flash -C tests/driver_itg320x BOARD=...
```

### CODE REVIEW

File: tests/driver_itg320x/main.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    #if USE_ITG320X_DRDY
+    /* init INT2/DRDY signal pin and enable the interrupt */
+    gpio_init_int(itg320x_params[0].int_pin, GPIO_IN, GPIO_RISING,
+                  itg320x_isr_data_ready, 0);
+    itg320x_enable_int(&dev, true);
+    #endif /* USE_ITG320X_DRDY */
+
+    while (1) {
+        #if USE_ITG320X_DRDY
```
Comment (benpicco) [inline tests/driver_itg320x/main.c]: Don't indent the preprocessor macros.

**Comment by benpicco** [comment]
I hooked it up to my WeAct-f411ce and it still works like a charm :)

File: drivers/itg320x/itg320x.c [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(data != NULL);
+    DEBUG_DEV("data=%p", dev, data);
+
+    itg320x_raw_data_t raw;
+
+    EXEC_RET(itg320x_read_raw (dev, &raw));
+
+    data->x = ((double)raw.x * _gyro_scale) * 1000;
```
[the remark this answers is not in the document]
Comment (gschorcht) [reply drivers/itg320x/itg320x.c]: Indeed.

**Comment by gschorcht** [comment]
@benpicco Thanks for reviewing. In the meantime, I have reworked the driver a bit. I made the following changes:

- Floating point arithmetics were replaced by integer arithmetics according to your suggestions.
- The interrupt version of the driver is only compiled in if the new pseudomodule `itg320x_int` is used.
- The interrupt pin is now initialized by the driver and not the application if pseudomodule `itg320x_int` is used. A new `itg320x_init_int` function has been added for this purpose.
- Function `set_reg_bit` could be removed by using register values as enum value definitions.
- Error codes have now negative values.
- Clock source can now be selected by parameters.
- Data ready function has been added for polling for new date.
- The resolution of angular rates was reduced from milli-degrees per seconds to tenths of a degree per second to fit into 16-bit integer values. This also matches the accuracy of the sensor better.

File: tests/driver_itg320x/main.c [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+#include "thread.h"
+#include "xtimer.h"
+
+#include "itg320x.h"
+#include "itg320x_params.h"
+
+#define ITG320X_SLEEP   (50 * US_PER_MS)
+
+kernel_pid_t p_main;
```
Comment (benpicco) [inline tests/driver_itg320x/main.c]:
```suggestion
static kernel_pid_t p_main;
```

File: tests/driver_itg320x/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+either
+
+- periodically fetching the data at a rate lower than the sensor's
+  output data rate (ODR), or
+- fetching the data when the data-ready interrupt is triggered.
+
+To use the latter approach, the according GPIO to which the sensor's
+**INT** output pin is connected has to be configured and initialized
+by the application. In this test application, this is done by defining
+```USE_ITG320X_DRDY``` and overrding the default configuration
```
Comment (benpicco) [inline tests/driver_itg320x/README.md]: It's `USEMODULE += itg320x_int` now.

File: drivers/include/itg320x.h [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    int16_t z; /**< angular rate z-axis as 16 bit two's complements (yaw) */
+
+} itg320x_raw_data_t;
+
+/**
+ * @brief   ITG320X device initialization parameters
+ */
+typedef struct {
+
+    unsigned  dev;   /**< I2C device (default I2C_DEV(0)) */
```
[the remark this answers is not in the document]
Comment (benpicco) [reply drivers/include/itg320x.h]: still valid.
(`i2c_t` is always typedefed to `unsigned` anyway)

[the remark this answers is not in the document]
Comment (gschorcht) [reply drivers/include/itg320x.h]: Hm, I was sure that I changed it :thinking:

File: sys/auto_init/saul/auto_init_itg320x.c [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+#include "assert.h"
+#include "log.h"
+#include "saul_reg.h"
+#include "itg320x.h"
+#include "itg320x_params.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define ITG320X_NUM    (sizeof(itg320x_params) / sizeof(itg320x_params[0]))
```
[the remark this answers is not in the document]
Comment (gschorcht) [reply sys/auto_init/saul/auto_init_itg320x.c]: Yes, of course. The initial version of the driver was provided before we had the `ARRAY_SIZE`.

File: tests/driver_itg320x/README.md (hunk 4 above)
Comment (gschorcht) [reply tests/driver_itg320x/README.md]: Ah yeah, I changed it in `main.c` and forgot it in `README.md` :sunglasses:

**Comment by benpicco** [comment]
Nice, still works like a charm - please squash!

**Comment by gschorcht** [comment]
@benpicco Thanks for reviewing and merging.

**Pull request merged**